library/firewall_lib.py

```diff
@@ -299,6 +301,29 @@ def main():
             "icmp_block_inversion, target or zone needs to be set"
         )
 
+    if state == "present" or state == "absent":
```

Comment:
What about using `present` to make sure that a zone exists exactly with the given settings? `enabled` and `disabled` can be used as is to make sure that an items exist or do not exist in the settings.